W3ChampionsChatService/Chats/ChatHub.cs

```diff
@@ -52,12 +52,44 @@ public async Task SendMessage(string message)
                     Context.Abort();
                 } else {
                     var chatMessage = new ChatMessage(user, trimmedMessage);
-                    _chatHistory.AddMessage(chatRoom, chatMessage);
-                    await Clients.Group(chatRoom).SendAsync("ReceiveMessage", chatMessage);
+                    if (!await processChatCommand(chatMessage))
+                    {
+                        _chatHistory.AddMessage(chatRoom, chatMessage);
+                        await Clients.Group(chatRoom).SendAsync("ReceiveMessage", chatMessage);
+                    }
                 }
             }
         }
 
+        /// <summary>
+        /// Processes chat commands and returns a boolean indicating whether a command was processed 
+        /// or the message should be sent as a normal message.
+        /// </summary>
+        /// <param name="message">The chat message to process.</param>
+        /// <returns>True if a command was processed, false otherwise.</returns>
+        private async Task<bool> processChatCommand(ChatMessage message)
```

Comment:
Nit: Seems like C# (and this project) uses PascalCase for functions

```suggestion
        private async Task<bool> ProcessChatCommand(ChatMessage message)
```